packages/flutter/lib/src/rendering/wrap.dart

```diff
@@ -519,7 +562,53 @@ class RenderWrap extends RenderBox
     int childCount = 0;
     RenderBox? child = firstChild;
     while (child != null) {
-      final Size childSize = layoutChild(child, childConstraints);
+      Size childSize;
+
+      final WrapParentData childParentData = child.parentData! as WrapParentData;
+      WrapFit fit = childParentData.fit;
+
+      if (mainAxisLimit.isInfinite) {
```

Comment:
That the provided fit is ignored in certain situations is ... surprising. It should at least be documented that that's what's happening and I wonder if we just want to assert and tell you that the given configuration is illegal in this environment.

Reply:
`tight` and `runTight` means "you must fill the wrap I assign you to".
Since this doesn't work for an unbound main axis, I've decided to downgrade it to "you may fill the wrap I assign you to". In my opinion that's similar to the behavior of `Center` (it centers in all limited axis). Therefore I think it's convenient and not surprising. Furthermore, `Wrap` is not really usable in an unbound main axis scenario, since every child will be placed in the first run, but it also allows it.
However, I've added documentation for this is d29c8da702.